### DISCUSSION TITLE
Endlosschleife Initial Setup

### DISCUSSION THREAD

**Discussion by AlteEnte** [jomjol/AI-on-the-edge-device#2253] [General] [answered]
Hallo,

nachdem ich das initiale Setup durchlaufe, alles korrekt eingerichtet habe, und den ESP neu starte, lande ich immer wieder im Initial Setup.

Kann mir jemand sagen was ich falsch mache?

Viele Dank!

**Comment by friedpa** [comment]
Ist in der Config.ini der Parameter SetupMode = false gesetzt (Settings->Configuration->Expert->Edit config.ini direct

**Reply by AlteEnte** [reply]
Es war tatsächlich auf "no" gesetzt, aber "false" bringt keine Änderung. Gefühlt bootet er auch nicht neu, sondern aktualisiert die Seite nur ...

**Comment by friedpa** [comment, marked as answer]
Wenn die Config.ini danach SetupMode = false zeigt den ESP32-CAM stromlos machen, sollte er son nicht neu booten.

**Reply by AlteEnte** [reply]
Das hat tatsächlich funktioniert. Der Reboot scheint bei mir nur durch stromlos machen zu funktionieren ...

**Reply by Slider0007** [reply]
Ich habe dazu ein Issue erstellt --> #2256
Das Reboot nach Setup Ende scheint tatsächlich nicht zu funtionieren.

**Comment by Slider0007** [comment]
Am Besten bei offenem Browserfenster auch einen Force Refresh mit CTRL+F5 durchführen, damit der Browsercache geleert wird.

**Comment by AlteEnte** [comment]
Ich hab auch das Gefühl, dass der ESP nicht das Blitzlicht benutzen mag. Kann ich das irgendwie testen?

**Reply by ghost** [reply]
Nach reboot hat einer meiner ESPs ein ähnliches Verhalten. Ab der zweiten Runde klappt es dann. Da sind aber die WS2812 dran.

**Comment by ghost** [comment]
Ich hatte am Wochenende ein ähnliches Phänomen nach Update von 12.0.1 auf 15.1.0 (komplett neu). Mir scheint eher, dass die Config vom Browser gecached wurde und immer wieder die initiale Seite angezeigt hat. http://<watermeter>/ zeigte dann aber die normale Seite.

**Comment by ARPIpappen** [comment]
Bei mir gestern auch so. Mein erster Wasserzähler konnte ich ohne Probleme einrichten, jetzt wollte ich den Stromzähler smart machen. Hier ständig Endlosschleife. Gibt es schon einen Trick?

**Comment by friedpa** [comment]
Es scheint so, dass der Reboot nach dem Setup nicht funktioniert. Die config.ini checken (wie oben beschrieben) und dann den "Stecker ziehen"

**Reply by caco3** [reply]
Das wird in https://github.com/jomjol/AI-on-the-edge-device/issues/2256 gefixt werden.

**Comment by JWRu** [comment]
Ich hatte ein ähnliches Problem: Konfiguration abgeschlossen ("save" 5 x kontrolliert). Nach dem Reboot landet man wieder auf der Konfigurationsseite. Auch spannungslos machen hat nicht geholfen. In der config.ini stand immer noch "SetupMode = true". Habe von Hand "false" eingetragen - seitdem läuft's.

**Reply by caco3** [reply]
Steht irgend etwas im Log? Könnte es sein, dass das Dateisystem nicht schreibbar ist?

**Comment by schneich** [comment]
gibt es eine Möglichkeit, von remote auf die SD-Karte zuzugreifen? Ich habe die config gerade zum vierten Mal gemacht. Wenn ich jetzt die SD-Karte herausnehmen muss, dann muss ich die config noch einmal machen... und die Kinder schlafen schon, da kann ich nicht dauernd in den Keller rennen. ;-)

**Reply by caco3** [reply]
Ja, siehe Menü `System > File Server`, da kannst Du Dateien löschen und hochladen. Es gibt auch eine Backup-Funktion `System > Backup/Restore`

**Reply by schneich** [reply]
Danke für die schnelle Antwort. Ich hänge in der aller ersten config fest und habe gar keine Menu-bar oben. Geht irgendwas übers Netzwerk? SSH? Oder kann ich via Browser auf eine andere Seite zugreifen?

**Reply by caco3** [reply]
Ja, das geht: `http://<IP>/backup.html` resp. `http://<IP>/fileserver/`. Beachte, dass der abschliessende `/` benötigt wird!
